p2panda-rs/src/operation/plain/plain_value.rs

```diff
@@ -124,10 +164,71 @@ impl From<Vec<DocumentViewId>> for PlainValue {
     }
 }
 
+/// Helper for converting a cbor value into a plain operation value.
+fn cbor_value_to_plain_value(value: Value) -> Result<PlainValue, PlainValueError> {
+    let result: Result<PlainValue, PlainValueError> = match value {
+        Value::Integer(int) => {
+            let int: i64 = int.try_into()?;
+            Ok(int.into())
+        }
+        Value::Bytes(bytes) => Ok(bytes.into()),
+        Value::Float(float) => Ok(float.into()),
+        Value::Text(text) => Ok(text.into()),
+        Value::Bool(bool) => Ok(bool.into()),
+        Value::Array(array) => cbor_array_to_plain_value_list(array),
+        _ => return Err(PlainValueError::UnsupportedValue),
+    };
+
+    result
+}
+
+/// Helper for converting a cbor array into a plain operation list value.
+///
+/// This method can fail which means the passed value is not an `AmbiguousRelation` or
+/// `PinnedRelation` plain value variant.
+fn cbor_array_to_plain_value_list(array: Vec<Value>) -> Result<PlainValue, PlainValueError> {
+    // First attempt to parse this vec of values into a vec of strings. If this succeeds it means
+    // this is an `AmbiguousRelation`
+    let ambiguous_relation: Result<Vec<String>, _> = array
+        .iter()
+        .map(|value| match value.as_text() {
+            Some(text) => Ok(text.to_string()),
+            None => Err(PlainValueError::UnsupportedValue),
+        })
+        .collect();
+
+    // If this was successful we stop here and return the value.
+    if let Ok(strings) = ambiguous_relation {
+        return Ok(strings.into());
```

Comment:
Great, this is much nicer! Thank you!!

A little detail, but how about writing tgis out to `AmbigiousRelation:: From`, then it gets very explicit what we're dealing with ✊🏻